internal/wasm/call_context.go

```diff
@@ -73,29 +73,27 @@ func (m *CallContext) FailIfClosed() (err error) {
 //
 // Callers of this function must invoke the returned context.CancelFunc to release the spawned Goroutine.
 func (m *CallContext) CloseModuleOnCanceledOrTimeout(ctx context.Context) context.CancelFunc {
-	goroutineDone, cancelFn := context.WithCancel(context.Background())
-	go m.closeModuleOnCanceledOrTimeoutClosure(ctx, goroutineDone)()
-	return cancelFn
+	cancelChan := make(chan struct{})
```

Comment:
maybe add a comment about why not context.WithCancel or if too long a blurb in RATIONALE.md? Seems took enough work to want to keep that info in case it gets "fixed" later ;)

Reply:
I guess basically the main thing is to say how this prevents arbitrary choice of the wrong branch in the select, not necessarily about cancelcontext, but in any case whichever part of this is critical to correctness would be a great place to comment/warn to prevent someone from undoing this later.

Reply:
For sure, added some comments. To be clear this bit didn't fix the bug, this was just to reduce the resources enabling `WithCloseOnContextDone` uses a bit.